dockerversion/useragent.go

```diff
@@ -8,8 +8,18 @@ import (
 )
 
 // DockerUserAgent is the User-Agent the Docker client uses to identify itself.
-// It is populated from version information of different components.
-func DockerUserAgent() string {
+// In accordance with RFC 7231 (5.5.3) is of the form:
+//    [upstream UA] [docker client's UA]
+// upstreamUA is optional
+func DockerUserAgent(upstreamUA ...string) string {
+	// ugliness to fake an optional arg in golang
+	var _upstreamUA string
```

Comment:
I would prefer to just have `DockerUserAgent` take a string argument, and callers can pass `""` if they don't have an upstream user agent to include. It makes it a little clearer how the function should be used, i.e. that it only takes at most one argument.

If you do keep the optional argument setup, please rename `_upstreamUA`. Underscores are discouraged in golang style, and might cause golint to complain.